src/components/four-up.tsx

```diff
@@ -85,54 +86,77 @@ export class FourUpComponent extends BaseComponent<IProps, {}> {
             });
             return {
               doc: groupUserDoc,
-              initials: groupUser.initials
+              initials: groupUser.initials,
+              name: groupUser.name
             };
           })
       : [];
 
     const groupDoc = (index: number) => {
-      const doc = groupUsers[index] && groupUsers[index].doc;
-      // Ghost users can see all documents
-      const showDoc = doc && (index === 0 || doc.visibility === "public" || isGhostUser);
-      return showDoc ? doc : undefined;
+      return groupUsers[index] && groupUsers[index].doc;
+    };
+
+    const hideCanvas = (index: number) => {
+      const doc = groupDoc(index);
+      const unopenedDoc = groupUsers[index] && !doc;
+      // Don't hide anything from ghost users, and treat unopened documents as private by default
+      return !isGhostUser && (unopenedDoc || doc && doc.visibility === "private");
     };
 
     // If the user is real, their document should be displayed first
     if (!isGhostUser) {
       groupUsers.unshift({
         doc: document,
-        initials: user.initials
+        initials: user.initials,
+        name: user.name
       });
     }
 
+    const nwCanvas = (
+      <CanvasComponent context="four-up-nw" scale={nwCell.scale}
+                       editabilityLocation={groupUsers[0] && "north west"}
+                       readOnly={isGhostUser /* Ghost users do not own group documents and cannot edit others' */}
+                       document={groupDoc(0)} {...others} />
+    );
+    const neCanvas = (
+      <CanvasComponent context="four-up-ne" scale={neCell.scale}
+                       editabilityLocation={groupUsers[1] && "north east"}
+                       readOnly={true} document={groupDoc(1)} {...others} />
+    );
+    const seCanvas = (
+      <CanvasComponent context="four-up-se" scale={seCell.scale}
+                       editabilityLocation={groupUsers[2] && "south east"}
+                       readOnly={true} document={groupDoc(2)} {...others}/>
+    );
+    const swCanvas = (
+      <CanvasComponent context="four-up-sw" scale={swCell.scale}
+                       editabilityLocation={groupUsers[3] && "south west"}
+                       readOnly={true} document={groupDoc(3)} {...others}/>
+    );
```

Comment:
Clearly we need a more elegant way of generating all of these, but I left it for now